src/20250120_Multi_FebruaryFundingUpdate/AaveV3Avalanche_FebruaryFundingUpdate_20250120.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {AaveV2Avalanche, AaveV2AvalancheAssets} from 'aave-address-book/AaveV2Avalanche.sol';
+import {AaveV3Avalanche, AaveV3AvalancheAssets} from 'aave-address-book/AaveV3Avalanche.sol';
+import {CollectorUtils, ICollector} from 'aave-helpers/src/CollectorUtils.sol';
+import {IProposalGenericExecutor} from 'aave-helpers/src/interfaces/IProposalGenericExecutor.sol';
+
+/**
+ * @title February Funding Update
+ * @author @TokenLogic
+ * - Snapshot: Direct-To-AIP
+ * - Discussion: https://governance.aave.com/t/arfc-february-funding-update/20712
+ */
+contract AaveV3Avalanche_FebruaryFundingUpdate_20250120 is IProposalGenericExecutor {
+  using CollectorUtils for ICollector;
+
+  uint256 public constant DAI_TO_WITHDRAW = 1_000_000 ether;
+
+  function execute() external {
+    _withdraw();
+    _deposit();
+  }
+
+  function _withdraw() internal {
+    AaveV3Avalanche.COLLECTOR.withdrawFromV2(
+      CollectorUtils.IOInput({
+        pool: address(AaveV2Avalanche.POOL),
+        underlying: AaveV2AvalancheAssets.WETHe_UNDERLYING,
+        amount: IERC20(AaveV2AvalancheAssets.WETHe_A_TOKEN).balanceOf(
+          address(AaveV3Avalanche.COLLECTOR)
+        ) - 1 ether
+      }),
+      address(AaveV3Avalanche.COLLECTOR)
+    );
+
+    AaveV3Avalanche.COLLECTOR.withdrawFromV2(
+      CollectorUtils.IOInput({
+        pool: address(AaveV2Avalanche.POOL),
+        underlying: AaveV2AvalancheAssets.DAIe_UNDERLYING,
```

Comment:
why 1M instead of all - 1 (as commented in the forum?)

Reply:
there is only 1.1M available to withdraw

Reply:
the availability keeps jumping up and down so we conservatively picked a number